components/brave_wallet_ui/ledger/ledger.ts

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { LedgerCommand, kLedgerBridgeUrl } from '../common/hardware/ledgerjs/ledger-messages'
+import { LedgerUntrustedMessagingTransport } from '../common/hardware/ledgerjs/ledger-untrusted-transport'
+
+const params = new Proxy(new URLSearchParams(window.location.search), {
```

Comment:
Why is a proxy needed?

Reply:
I didn't put much thought into this, really just replicating an approach in this pull request https://github.com/brave/brave-core/pull/13370/files#diff-8109eea81b56903719f0c660dd59c76048b43c0cd1759d4f2df592924089bc95R6.